File: backend/src/main/java/com/example/backend/Service/projet/ProjetService.java
package com.example.backend.Service.projet;

import com.example.backend.exception.ResourceAlreadyExistException;
import com.example.backend.exception.ResourceNotFoundException;
import com.example.backend.model.Projet;
import com.example.backend.repository.ProjetRepo;
import com.example.backend.request.ProjetRequest;
import lombok.RequiredArgsConstructor;
import org.springframework.stereotype.Service;

import java.util.List;

@Service
@RequiredArgsConstructor
public class ProjetService implements IProjetService{
    private final ProjetRepo projetRepo;
    @Override
    public List<Projet> getAllProjets() {
        return projetRepo.findAll();
    }

    @Override
    public Projet addProjet(ProjetRequest projet) {
        Projet existingProject = projetRepo.findByNom(projet.getNom());
        if (existingProject != null) {
            throw new ResourceAlreadyExistException("Le projet existe déjà !");
        }

        Projet newProject = new Projet();
        newProject.setNom(projet.getNom());
        newProject.setDescription(projet.getDescription());

        return projetRepo.save(newProject);
    }


    @Override
    public Projet updateProjet(ProjetRequest projet, Long id) {
        Projet projetFound = projetRepo.findById(id)
                .orElseThrow(() -> new ResourceNotFoundException("Le projet non trouvé"));
        projetFound.setNom(projet.getNom());
        projetFound.setDescription(projet.getDescription());
        projetRepo.save(projetFound);
        return projetFound;
    }

    @Override
    public void deleteProjet(Long id) {
        Projet projetFound = projetRepo.findById(id)
                .orElseThrow(() -> new ResourceNotFoundException("Le projet non trouvé"));
        projetRepo.delete(projetFound);
    }

    @Override
    public Projet getProjetById(Long id) {
        return projetRepo.findById(id)
                .orElseThrow(() -> new ResourceNotFoundException("Le projet non trouvé"));
    }
}

File: frontend/services/projetService.js
import { authorizedApi } from "../utils/api";

export const getAllProjets = async () => {
  const response = await authorizedApi().get("/projet");
  return response.data;
};

export const getProjetById = async (id) => {
  const response = await authorizedApi().get(`/projet/${id}`);
  return response.data;
};

export const createProjet = async (data) => {
  const response = await authorizedApi().post("/projet", data);
  return response.data;
};

export const updateProjet = async (data) => {
  const response = await authorizedApi().put("/projet/", data);
  return response.data;
};

export const deleteProjet = async (id) => {
  const response = await authorizedApi().delete(`/projet/${id}`);
  return response.data;
};
